### DISCUSSION TITLE
Define API - Open in dev box?

### DISCUSSION THREAD

**Discussion by Porter-smith** [codesandbox/codesandbox-client#8454] [Q&A] [answered]
I'm trying to use the CodeSandbox API to launch my project in a 'dev box' environment using a `devcontainer.json` file, as I've seen suggested in several posts. Despite following these suggestions, my project still opens in a standard sandbox environment. There isn’t much documentation on this specific setup, so I'm unsure what might be missing or incorrect in my approach. Could anyone help me figure out why it's not launching in a 'dev box' environment and how to set it up correctly? Here's the link to my project setup: [CodeSandbox Project](https://codesandbox.io/p/sandbox/define-api-post-request-forked-kzww8h?file=%2Findex.js). Thanks in advance for your help!

**Comment by filipelima18** [comment, marked as answer]
Hi @Porter-smith. As stated in our docs, our Define API currently only supports Sandboxes, not Devboxes.

**Reply by Porter-smith** [reply]
oh okay ! thank you!

**Reply by RSS1102** [reply]
> Hi @Porter-smith. As stated in our docs, our Define API currently only supports Sandboxes, not Devboxes.

But does the documentation tell us that we can use the `environment: "server"` api to create a devbox?

https://codesandbox.io/docs/learn/devboxes/your-first-sandbox#programmatically-creating-devboxes

**Reply by filipelima18** [reply]
Hi @RSS1102 - I'm verifying this with the team to understand if it's just a matter of updating the Docs or if there were any changes to the API that affect Devboxes.

**Reply by prashanth726** [reply]
@filipelima18 any update on this, thanks

**Reply by filipelima18** [reply]
@prashanth726 We recently released CodeSandbox SDK, which contains a new set of APIs specifically for managing Devboxes. Please refer to those [Docs](https://codesandbox.io/docs/sdk).
